diff -Naur source-old/src/CMakeLists.txt source-new/src/CMakeLists.txt
--- source-old/src/CMakeLists.txt	2023-09-11 21:48:10.004961626 +0300
+++ source-new/src/CMakeLists.txt	2023-09-11 21:44:31.915689255 +0300
@@ -358,12 +358,7 @@
 		set(LUA_TAG "luajit51")
 		if (EXISTS ${EXTERNAL_SRC_DIR}/git/luajit)
 			ExternalProject_Add(luajitbuild
-				SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/luajit
-				GIT_REPOSITORY "${EXTERNAL_SRC_DIR}/git/luajit"
-				CONFIGURE_COMMAND ""
-				GIT_TAG "v2.1.0-beta3"
-				UPDATE_COMMAND ""
-				INSTALL_COMMAND ""
+ 				SOURCE_DIR "${EXTERNAL_SRC_DIR}/git/luajit"
 				BUILD_IN_SOURCE 1
 				BUILD_COMMAND "${EXTMAKE_CMD}"
 				DEFAULT_CC=${CMAKE_C_COMPILER}
